### PR TITLE
Modulo operator support

### PR DESCRIPTION
**Pull request by boki1**
Related to Issue #10 

------

# Description

Currently no modulo (`%`) operator is provided, so add that to the binary arithmetic operations supported by the compiler.

# Task

Provide implementation of the modulo (`%`) operator where required. 

# Definition of Done

A successful build of the gcd algorithm.

### CODE REVIEW

File: test/test-this-output.txt [hunk 1]
```diff
@@ -0,0 +1,28 @@
+----------------------------------
```
Comment (vkabadzhova) [inline test/test-this-output.txt]: I'm not sure if these files should be commited. Maybe it is a good idea to have an example output, but I prefer to put it in the Wiki, not commit it. Please, take this into account.

File: test/test.sh [hunk 2]
```diff
@@ -22,17 +25,18 @@ do
...
+# Generate correct output and store it in `CORRECT_OUTPUT`
+./"$TEST_EXE_FILE" "$TEST_EXE_PARAMS" >> "$OUTPUT" 2>&1
 
+# Compare `THIS_OUTPUT` and `CORRECT_OUTPUT` and store it in `OUTPUT`
 diff -sy --width=120 "$CORRECT_OUTPUT" "$THIS_OUTPUT" > "$OUTPUT"
 
-rm "$THIS_OUTPUT"
-rm "$CORRECT_OUTPUT"
-less "$OUTPUT"
+# rm "$THIS_OUTPUT"
```
Comment (vkabadzhova) [inline test/test.sh]: Delete comments and make some clean in the Makefile

File: .gitmodules [hunk 3]
```diff
@@ -1,6 +1,3 @@
-[submodule "gvpp"]
```
Comment (vkabadzhova) [inline .gitmodules]: Thank you!

File: src/main.cpp [hunk 4]
```diff
@@ -5,6 +5,8 @@
 #include "include/parser.hpp"
 #include "include/parsing_visualizer.hpp"
 
+#define PARSING_VISUALIZATION_ENABLED
```
Comment (vkabadzhova) [inline src/main.cpp:8]: OK, but should be fixed in next commits. We won't merge until this is already fixed

File: test/testcases/gcd.kv [hunk 5]
```diff
@@ -0,0 +1,11 @@
+gcd a, b:
```
Comment (vkabadzhova) [inline test/testcases/gcd.kv:1]: Yes! :100:

File: test/testcases/fib.kv [hunk 6]
```diff
@@ -0,0 +1,10 @@
+fib n:
```
Comment (vkabadzhova) [inline test/testcases/fib.kv:1]: Yes again! :100: :partying_face:

**Review by vkabadzhova — CHANGES_REQUESTED** [review]
First, sorry for the single comments that are reviews, it looks like I've really messed something with the reviewing order and now I only can put reviews, so just resolve them. Then good job, I like the work. I have 2 things I insist to be changed before we merge:
- [x] Remove the sample outputs in the `test/` directory and if you want samples, put them in the documentation
- [x] Please, just try to fix the visualization flag. It should not be always enabled.

File: src/main.cpp (hunk 4 above)
Comment (boki1) [reply src/main.cpp:8]: This flag is set only for demonstration purposes. In future releases an argument parser would be a really good addition. So would you be OK, if this is remains unchanged for now.

Comment (vkabadzhova) [reply src/main.cpp:8]: OK

File: test/test-this-output.txt (hunk 1 above)
Comment (boki1) [reply test/test-this-output.txt]: Fixed it.

File: test/test.sh (hunk 2 above)
Comment (boki1) [reply test/test.sh]: Fixed it.

**Review by vkabadzhova — APPROVED** [review]

**Review by vkabadzhova — APPROVED** [review]

**Pull request merged**